### DISCUSSION TITLE
posts/Go/consul/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#312] [Announcements]
https://www.liwenzhou.com/posts/Go/consul/ 

 李文周的Blog提供免费的Go语言学习教程，本文介绍了基于consul实现服务注册与发现。

**Comment by munding** [comment]
consul 可以作为配置中心吗？好像没看到有使用的

**Reply by Q1mi** [reply]
可以但不推荐。

**Reply by evanzme** [reply]
想问下为什么不推荐呢？

**Comment by estrella-zys** [comment]
你好，我在实现的时候出现了健康检查失败的情况，请问什么原因会导致出现这个问题呢

**Reply by Q1mi** [reply]
consul client 和你的服务之间不通。

**Comment by Drengin6306** [comment]
在使用mTLS加密的情况下，api.AgentServiceCheck中可以配置useTLS和servername，但是无法配置证书，导致无法进行健康检查，这个问题怎么解决
